### PR TITLE
Add UUID generator module

### PR DESCRIPTION
**Pull request by g-despot**
A very simple query module that generates string UUIDs.

### CODE REVIEW

File: cpp/uuid_module/uuid_module.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    if (record == nullptr) {
+      throw mg_exception::NotEnoughMemoryException();
+    }
+
+    mg_utility::InsertStringValueResult(record, kFieldUuid, string, memory);
+  } catch (const std::exception &e) {
+    // We must not let any exceptions out of our module.
+    mgp_result_set_error_msg(result, e.what());
+    return;
+  }
```
Comment (jmatak) [inline cpp/uuid_module/uuid_module.cpp:28]: Is the goal of this module to obtain a uuid for each node in graph?

Comment (g-despot) [reply cpp/uuid_module/uuid_module.cpp:28]: The goal is to return a newly generated string UUID just like in [this](https://neo4j.com/labs/apoc/4.1/graph-updates/uuid/#manual-uuids) APOC procedure.

Comment (jmatak) [reply cpp/uuid_module/uuid_module.cpp:28]: Alright that's cool, please write an e2e test before merging this to `main`

Comment (g-despot) [reply cpp/uuid_module/uuid_module.cpp:28]: The e2e test is written and committed, you can try it out.

**Review by jmatak — APPROVED** [review]

**Pull request merged**